cpp/tensorrt_llm/thop/groupRmsNormOp.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tensorrt_llm/common/cudaUtils.h"
+#include "tensorrt_llm/common/dataType.h"
+#include "tensorrt_llm/common/opUtils.h"
+#include "tensorrt_llm/kernels/groupRmsNormKernels/groupRmsNormKernels.h"
+#include "tensorrt_llm/runtime/torchUtils.h"
+#include "tensorrt_llm/thop/thUtils.h"
+
+#include <ATen/cuda/CUDAContext.h>
+#include <cstdint>
+#include <cuda_runtime.h>
+#include <stdexcept>
+#include <torch/extension.h>
+#include <vector>
+
+namespace torch_ext
+{
+
+void groupRMSNorm(torch::TensorList const& inputs, torch::TensorList const& outputs, torch::TensorList const& weights,
+    double eps, double weight_bias, bool enable_weights)
+{
+    TORCH_CHECK(inputs.size() > 0, "Input tensor list cannot be empty.");
+    auto const first_input = inputs[0];
+    auto const dtype = first_input.scalar_type();
+    auto const device = first_input.device();
+    uint32_t const num_inputs = inputs.size();
+    TORCH_CHECK(num_inputs <= 2, "Only up to 2 inputs are supported.");
+    uint32_t const batch_size = first_input.sizes()[0];
+
+    TORCH_CHECK(first_input.dim() == 2, "Inputs must be 2D tensors [batch_size, hidden_dim].");
+    TORCH_CHECK(first_input.is_cuda(), "Inputs must be CUDA tensors.");
+
+    if (enable_weights)
+    {
+        TORCH_CHECK(weights.size() == num_inputs, "Weights list size must match inputs list size.");
+    }
+
+    for (size_t i = 0; i < num_inputs; ++i)
+    {
+        TORCH_CHECK(inputs[i].sizes()[0] == batch_size, "Inputs must have the same batch size.");
+        TORCH_CHECK(inputs[i].dim() == 2, "Inputs must be 2D tensors [batch_size, hidden_dim].");
+        TORCH_CHECK(inputs[i].device() == device, "Inputs must be on the same device.");
+        TORCH_CHECK(inputs[i].scalar_type() == dtype, "Inputs must be of the same type.");
+        TORCH_CHECK(
+            inputs[i].sizes()[1] == weights[i].sizes()[0], "Inputs and weights must have the same last dimension.");
+    }
+
+#define DISPATCH_INPUT_SIZES(n)                                                                                        \
+    case n:                                                                                                            \
+    {                                                                                                                  \
```

Comment:
```suggestion
#define DISPATCH_INPUT_SIZES(n)                                                                                        \
    {                                                                                                                  \
```

Do not include `case` in the macro. Leave it out

Reply:
Good point!